### ISSUE TITLE
Fix testing method

### ISSUE THREAD

**Issue by Velua** [bancorprotocol/webapp#464]

**Comment by sirius651** [comment]
test

**Comment by Velua** [comment]
@sirius651 

Let's 

- [ ] Update testing so that it supports ```.ts``` files instead of the current ```.js``` ones
- [ ] Confirm direct imports of functions into these test files works okay. 
- [ ] Come up with a component testing solution.
- [ ] Bonus: Any Vuex related testing libraries? What's the most professional way for assessing it.

**Comment by sirius651** [comment]
The current situation I have approached for VueX and component testing is to remove the existing coupled vuex-class-component and make sure it can be replaced.
First off, I used with pure vuex modules such as Module, GetterTree, MutationTree, ActionTree. but, it made a boilerplate code a bit.
So, I reached out that as a good combination, vuex uses vuex-module-decorator, and vue-class in components.
With this, we can write a testable code and mocking store is also possible. Also, to implement vuex store is very similar to the current code method.
